### PR TITLE
Add STM32H7 family

### PR DESCRIPTION
**Pull request by AlyoshaVasilieva**
All chips except 7A3/7B3/7B0 use the same addresses.

`stm32h7` covers: RM0468 (723/733, 725/735, 730), RM0433 (742, 743/753, 750), and RM0399 (745/755, 747/757).

`stm32h7bx` covers: RM0455 (7A3/7B3, 7B0).

Open to suggestions on feature naming.

Tested on an STM32H750VBT6 and an STM32H743VIT6. Appears to work; the flash size is certainly reading correctly.

Additionally add `.idea` dir to gitignore since my IDE puts garbage files in there.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /target
 Cargo.lock
+/.idea
```
Comment (adamgreig) [inline .gitignore]: This doesn't belong in a per-project `.gitignore`; you might consider adding it to your [global git ignore instead](https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/).

**Review by adamgreig — CHANGES_REQUESTED** [review]
Names seem good enough; it's a shame ST don't have any nomenclature for the A/B parts but I couldn't find anything. The only other option I can think of is adding a bunch of duplicate features `stm32h72x`, `stm32h73x`, `stm32h74x`, and `stm32h75x`, and then using `#[cfg(any(feature="stm32h72x", feature="stm32h73x", ...)]`. Probably also add `stm32h7ax` at that point. It's a bit more of a pain for us but might be a bit less confusing, what do you think?

File: .gitignore (hunk 1 above)
Comment (AlyoshaVasilieva) [reply .gitignore]: I had no idea that existed, thanks.

**Comment by AlyoshaVasilieva** [comment]
> Names seem good enough; it's a shame ST don't have any nomenclature for the A/B parts but I couldn't find anything. The only other option I can think of is adding a bunch of duplicate features `stm32h72x`, `stm32h73x`, `stm32h74x`, and `stm32h75x`, and then using `#[cfg(any(feature="stm32h72x", feature="stm32h73x", ...)]`. Probably also add `stm32h7ax` at that point. It's a bit more of a pain for us but might be a bit less confusing, what do you think?

I'm not a huge fan of multiple features leading to the same implementation, since it might make a user believe there is an underlying difference here. Not a strong opinion though, I'll switch to that definition method if you'd prefer it. (edit: Though a dupe feature for the A/B parts makes sense, due to the lack of a properly common name)

**Comment by adamgreig** [comment]
Having another feature for `stm32h7ax` and then using `stm32h7` for the rest sounds good to me!

**Comment by Disasm** [comment]
Thank you for the PR!
I'd better use Adam's naming suggestion with duplicate features. A single `stm32h7` feature that covers *some* of the subfamilies is confusing and doesn't look like a good idea. Also we already have specific subfamily features for F7 (`stm32f72x` and `stm32f73x`).

**Review by Disasm — APPROVED** [review]
Thank you!

**Comment by Disasm** [comment]
Released in v0.3.2 :tada:

**Pull request merged**